tests/carbon_offset_test.go

```diff
@@ -0,0 +1,85 @@
+package easypost_test
+
+import (
+	"github.com/EasyPost/easypost-go/v2"
+	"reflect"
+)
+
+func (c *ClientTests) TestShipmentCreateWithCarbonOffset() {
+	client := c.TestClient()
+	assert, require := c.Assert(), c.Require()
+
+	shipment, err := client.CreateShipmentWithCarbonOffset(c.fixture.BasicCarbonOffsetShipment())
+	require.NoError(err)
+
+	assert.Equal(reflect.TypeOf(&easypost.Shipment{}), reflect.TypeOf(shipment))
+	assert.NotNil(shipment.Rates)
+
+	rate := shipment.Rates[0]
+	assert.NotNil(rate.CarbonOffset)
+}
+
+func (c *ClientTests) TestShipmentBuyWithCarbonOffset() {
+	client := c.TestClient()
+	assert, require := c.Assert(), c.Require()
+
+	shipment, err := client.CreateShipment(c.fixture.FullCarbonOffsetShipment())
+	require.NoError(err)
+
+	lowestRate, err := client.LowestShipmentRate(shipment)
+	require.NoError(err)
+
+	boughtShipment, err := client.BuyShipmentWithCarbonOffset(shipment.ID, &lowestRate, "")
+	require.NoError(err)
+
+	assert.NotNil(boughtShipment.PostageLabel)
+
+	assert.NotNil(boughtShipment.Fees)
+	carbonOffsetExists := false
+	for _, fee := range boughtShipment.Fees {
+		if fee.Type == "CarbonOffsetFee" {
+			carbonOffsetExists = true
+		}
+	}
+	assert.True(carbonOffsetExists)
+}
+
+func (c *ClientTests) TestShipmentOneCallBuyWithCarbonOffset() {
+	client := c.TestClient()
+	assert, require := c.Assert(), c.Require()
+
+	shipment, err := client.CreateShipment(c.fixture.OneCallBuyCarbonOffsetShipment())
+	require.NoError(err)
+
+	rates, err := client.RerateShipment(shipment.ID)
+	require.NoError(err)
+
+	assert.Equal(reflect.TypeOf([]*easypost.Rate{}), reflect.TypeOf(rates))
+	for _, rate := range rates {
+		assert.Equal(reflect.TypeOf(&easypost.Rate{}), reflect.TypeOf(rate))
+	}
```

Comment:
We aren't testing carbon offset at all here. Please have this match the Java assertions for this test.

Reply:
Good catch

Reply:
Copy-pasta resulted in the wrong thing getting deleted. Resolved.